### ISSUE TITLE
macOS test on CI can't find Catch

### ISSUE THREAD

**Issue by knelli2** [sxs-collaboration/spectre#3984]
# Bug reports:

Appears that cmake can't find Catch. This has happened on multiple of my PRs recently.

First appeared in https://github.com/sxs-collaboration/spectre/actions/runs/2348962964 from #3982 (however, this clearly didn't cause it). Has also appeared in https://github.com/sxs-collaboration/spectre/actions/runs/2349526892, https://github.com/sxs-collaboration/spectre/actions/runs/2353675414, and https://github.com/sxs-collaboration/spectre/actions/runs/2354054138.

Here's the error from the "Configure build with cmake" phase of the macOS test. The logs from each of these CI runs indicate that the spectre dependencies were, in fact, installed (including Catch).

```
CMake Error at /usr/local/Cellar/cmake/3.23.1_1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Catch (missing: CATCH_INCLUDE_DIR) (Required is at least
  version "2.8.0")
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.23.1_1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindCatch.cmake:45 (find_package_handle_standard_args)
  cmake/SetupCatch.cmake:4 (find_package)
  CMakeLists.txt:147 (include)


-- Configuring incomplete, errors occurred!
See also "/Users/runner/work/spectre/spectre/build/CMakeFiles/CMakeOutput.log".
See also "/Users/runner/work/spectre/spectre/build/CMakeFiles/CMakeError.log".
Error: Process completed with exit code 1.
```

**Comment by nilsvu** [comment]
I believe Catch2 v3 was released and gets picked up by the CI workflow now. Should be an easy fix, just constrain to `catch2@2.8:2` in `support/DevEnvironments/spack.yaml`.

**Comment by knelli2** [comment]
Yes I believe that fixed it. Thanks!
